src/app/submission/pages/submission.component.ts

```diff
@@ -277,31 +319,44 @@ export class SubmissionComponent implements OnInit, OnDestroy {
     return links && links[linkName] ? links[linkName]['href'] : null;
   }
 
-  downloadFile() {
-    this.downloadDisabled = true;
+  generateSpreadsheet() {
+    const uuid = this.submissionEnvelope['uuid']['uuid'];
+    this.brokerService.generateSpreadsheetFromSubmission(uuid)
+      .subscribe(response => {
+          this.alertService.success('',
+            'Successfully triggered spreadsheet generation! This may take a while. ' +
+            'Please come back later and check the link to download the file.');
+          this.downloadJobOngoing = true;
+        },
+        error => {
+          const err = 'An error occurred in the request to generated the spreadsheet'
+          this.alertService.error('Error', err);
+          console.error(err, error);
+        });
+  }
+
+  downloadSpreadsheet() {
     const uuid = this.submissionEnvelope['uuid']['uuid'];
     this.loaderService.display(true, 'This may take a moment. Please wait...');
-    // now set a cookie to disable further download attempt within X min of this request or
-    // until a response is returned when the cookie gets deleted.
-    this.cookieService.set(this.DOWNLOAD_TIMEOUT_COOKIE, "1", {expires: this.DOWNLOAD_BACKOFF_MINS / (24*60)})
+
     this.brokerService.downloadSpreadsheet(uuid).subscribe(response => {
-      const filename = response['filename'];
-      const newBlob = new Blob([response['data']]);
+        const filename = response['filename'];
+        const newBlob = new Blob([response['data']]);
         this.saveFileService.saveFile(newBlob, filename);
         this.loaderService.display(false);
-      this.downloadDisabled = false;
-      this.cookieService.delete(this.DOWNLOAD_TIMEOUT_COOKIE);
-    },
-    err => {
-      if (err instanceof TimeoutError) {
+      },
+      err => {
+        const retry_message = 'Please retry later.';
+        if (err instanceof TimeoutError) {
+          this.alertService.error('', 'Spreadsheet download timed out. ' + retry_message, false, true);
+        } else {
+          console.error(err)
```

Comment:
Do we need to log the error to the console?

Reply:
Might be helpful to log the full error object if there's an issue. I'm not sure what's the ideal way to report client errors.